From 6e9a19c43acac7d4254910906329d98d7b59085a Mon Sep 17 00:00:00 2001
From: Ren Yu <yux.ren@intel.com>
Date: Fri, 24 May 2019 14:55:10 +0800
Subject: [PATCH] Add the pre-timeout interrupt defined in IPMI spec

The IPMI watchdog pre-timeout interrupt is used to set the different
pre-timeout interrupt source. Add them as a dbus property,
IPMI set/get watchdog commands will use it.

Signed-off-by: Ren Yu <yux.ren@intel.com>
---
 xyz/openbmc_project/State/Watchdog.interface.yaml | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/xyz/openbmc_project/State/Watchdog.interface.yaml b/xyz/openbmc_project/State/Watchdog.interface.yaml
index 2fc47d8..6dfa9b9 100644
--- a/xyz/openbmc_project/State/Watchdog.interface.yaml
+++ b/xyz/openbmc_project/State/Watchdog.interface.yaml
@@ -33,6 +33,11 @@ properties:
       description: >
           The action the watchdog should perform when it expires.
       default: 'HardReset'
+    - name: PreTimeoutInterrupt
+      type: enum[self.PreTimeoutInterruptAction]
+      description: >
+          The BMC generates the selected interrupt before the timer expires.
+      default: 'None'
     - name: Interval
       type: uint64
       description: >
@@ -73,6 +78,23 @@ enumerations:
          description: >
            Perform a power cycle of the system.
 
+   - name: PreTimeoutInterruptAction
+     description: >
+       The type of PreTimeout Interrupt.
+     values:
+       - name: 'None'
+         description: >
+           Do nothing.
+       - name: 'SMI'
+         description: >
+           SMI.
+       - name: 'NMI'
+         description: >
+           NMI / Diagnostic Interrupt.
+       - name: 'MI'
+         description: >
+           Messaging Interrupt.
+
    - name: TimerUse
      description: >
        The type of timer use.
-- 
2.7.4

